cli/actions/demo.go

```diff
@@ -0,0 +1,189 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/kubeshop/tracetest/cli/file"
+	"github.com/kubeshop/tracetest/cli/openapi"
+	"github.com/kubeshop/tracetest/cli/utils"
+	"github.com/mitchellh/mapstructure"
+	"go.uber.org/zap"
+)
+
+type demoActions struct {
+	resourceArgs
+}
+
+var _ ResourceActions = &demoActions{}
+
+func NewDemoActions(options ...ResourceArgsOption) demoActions {
+	args := NewResourceArgs(options...)
+
+	return demoActions{
+		resourceArgs: args,
+	}
+}
+
+func (demo demoActions) Apply(ctx context.Context, args ApplyArgs) error {
+	if args.File == "" {
+		return fmt.Errorf("you must specify a file to be applied")
+	}
+
+	demo.logger.Debug(
+		"applying demo",
+		zap.String("file", args.File),
+	)
+
+	fileContent, err := file.ReadRaw(args.File)
+	if err != nil {
+		return fmt.Errorf("could not read file: %w", err)
+	}
+
+	if fileContent.Definition().Type != "Demo" {
+		return fmt.Errorf(`file must be of type "Demo"`)
+	}
+
+	var demoResource openapi.Demo
+	mapstructure.Decode(fileContent.Definition().Spec, &demoResource.Spec)
+
+	if demoResource.Spec.Id == nil || *demoResource.Spec.Id == "" {
+		return demo.create(ctx, fileContent)
+	}
+
+	return demo.update(ctx, fileContent, *demoResource.Spec.Id)
+}
+
+func (demo demoActions) create(ctx context.Context, file file.File) error {
+	request, err := demo.resourceClient.NewRequest(demo.resourceClient.BaseUrl, http.MethodPost, file.Contents())
+	if err != nil {
+		return fmt.Errorf("could not create request: %w", err)
+	}
+
+	resp, err := demo.resourceClient.Client.Do(request)
+	if err != nil {
+		return fmt.Errorf("could not send request: %w", err)
+	}
```

Comment:
it's important to schedule a function to close the response body after you ensure you got no errors from it:

```suggestion
	resp, err := demo.resourceClient.Client.Do(request)
	if err != nil {
		return fmt.Errorf("could not send request: %w", err)
	}
	
	defer resp.Body.Close()
```